src/common/board.rs

```diff
@@ -31,23 +45,23 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
     }
 }
 
-pub struct FakeBoard {
-    analogs: Vec<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>,
-    i2cs: HashMap<String, Arc<Mutex<FakeI2CHandle>>>,
-    pin_pwms: HashMap<i32, f64>,
-    pin_pwm_freq: HashMap<i32, u64>,
-}
+/// Represents the functionality of a general purpose compute board that contains various components such as analog readers and digital interrupts.
 pub trait Board: Status {
+    /// Set a pin to high or low
     fn set_gpio_pin_level(&mut self, pin: i32, is_high: bool) -> anyhow::Result<()>;
+    /// Return the current [BoardStatus] of the board
```

Comment:
Very minor, but could these be offset by a blank line so the code and comments don't run together? Or will the rust formatter just shove them backtogether? Personally, I find it much more readable as:

```
    /// Set a pin to high or low
    fn set_gpio_pin_level(&mut self, pin: i32, is_high: bool) -> anyhow::Result<()>;

    /// Return the current [BoardStatus] of the board
    fn get_board_status(&self) -> anyhow::Result<common::v1::BoardStatus>;
```

It seems, at least, to be the style within the rust std lib: https://doc.rust-lang.org/src/core/alloc/mod.rs.html#164

Reply:
Sure!

We may want to look at adding a `rustfmt.toml` file in the future to enforce these things. 

Though it seems this specifically is still unstable. [Link](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#blank_lines_lower_bound)